src/serde.rs

```diff
@@ -65,3 +65,59 @@ impl<'de, K, V, S> Deserialize<'de> for OrderMap<K, V, S>
         deserializer.deserialize_map(OrderMapVisitor(PhantomData))
     }
 }
+
+
+use OrderSet;
+
+/// Requires crate feature `"serde-1"`
+impl<T, S> Serialize for OrderSet<T, S>
+    where T: Serialize + Hash + Eq,
+          S: BuildHasher
+{
+    fn serialize<Se>(&self, serializer: Se) -> Result<Se::Ok, Se::Error>
+        where Se: Serializer
+    {
+        let mut set_serializer = try!(serializer.serialize_seq(Some(self.len())));
+        for value in self {
+            try!(set_serializer.serialize_element(value));
+        }
+        set_serializer.end()
+    }
+}
+
+struct OrderSetVisitor<T, S>(PhantomData<(T, S)>);
+
+impl<'de, T, S> Visitor<'de> for OrderSetVisitor<T, S>
+    where T: Deserialize<'de> + Eq + Hash,
+          S: Default + BuildHasher
+{
+    type Value = OrderSet<T, S>;
+
+    fn expecting(&self, formatter: &mut Formatter) -> fmt::Result {
+        write!(formatter, "a set")
+    }
+
+    fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+        where A: SeqAccess<'de>
+    {
+        let mut values = OrderSet::with_capacity_and_hasher(seq.size_hint().unwrap_or(0), Default::default());
```

Comment:
`S::default()` is the neater style but I see it's just copying the existing code, so that's understandable.

Reply:
I updated both cases.